main.py

```
@@ -0,0 +1,1832 @@
+            # Asyncer Scan
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            
+            scan_result = loop.run_until_complete(
+                self.scanner.perform_scan(categories, progress_callback)
+            )
+            
+            # UI-Update im Hauptthread
+            self.root.after(0, lambda: self._on_scan_complete(scan_result))
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)